myproject.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:io'; // to add input,output
+
+void main() {
+  var noteApp = NoteTakingApp(); //Create an instance of the NoteTakingApp
+  var optione;
+
+  do {
+    noteApp.DisplayManu();
+    print("Enter a number from 1 to 5:");
+    optione = int.tryParse(stdin.readLineSync()!) ?? 0;
+
+    try {
+      switch (optione) {
+        // using swittch to evaluate the optione statment
+        case 1:
+          noteApp.Create();
+          break;
+
+        case 2:
+          noteApp.Edit();
+          break;
+
+        case 3:
+          noteApp.Delete();
+          break;
+
+        case 4:
+          noteApp.Search();
+          break;
+
+        case 5:
+          print("Exiting Note Taking App.");
+          print("---------------------------");
+          break;
+
+        default:
+          print("Invalid choice. please enter a number between 1 to 5.");
+          break;
+      }
+    } catch (e) {
+      print("An error acored: $e");
+    }
+  } while (optione != 5);
+}
+
+class NoteTakingApp {
```

Comment:
separating the user interface logic from the application logic. This can make your code more modular and easier to maintain.